### PR TITLE
8.0 mobile app inventory small fixes

### PR DESCRIPTION
**Pull request by hparfr**
@legalsylvain Please have a look

### CODE REVIEW

File: mobile_app_inventory/static/www/app.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ angular.module(
 
     $ionicConfigProvider.views.transition('none');
 
-    $urlRouterProvider.otherwise('/login');
+    $urlRouterProvider.otherwise('/inventory/');
```
Comment (legalsylvain) [inline mobile_app_inventory/static/www/app.js:81]: Sorry but I reverted this change you introduced during the port of this module.
This was creating error if user was not correctly logged. Did you fixed the problem ?

File: mobile_app_inventory/static/www/controllers/main_scan.js [hunk 2]
```diff
@@ -56,11 +56,14 @@ angular.module('mobile_app_inventory').controller(
...
-                scan_state.set_location(location);
-            });
+
             InventoryModel.get_inventory(toParams.inventory_id).then(function(inventory) {
                 scan_state.set_inventory(inventory);
+            }).then(function () {
+                //load location after inventory
+                LocationModel.get_location(toParams.location_id).then(function (location) {
+                    scan_state.set_location(location);
+                });
```
Comment (legalsylvain) [inline mobile_app_inventory/static/www/controllers/main_scan.js:66]: LGTM.

File: mobile_app_inventory/static/www/models/tools.js [hunk 3]
```diff
@@ -2,22 +2,19 @@
...
-                }, 10);
+            return promise.then(function (mode) {
+                var nextPage = {
+                    'one_page': 'main_scan',
+                    'automate': 'product'
+                }[mode];
+                return $state.go(nextPage, {
+                    inventory_id: inventory_id,
+                    location_id: location_id,
+                });
```
Comment (legalsylvain) [inline mobile_app_inventory/static/www/models/tools.js:17]: Very much synthetic ! :+1:

File: mobile_app_inventory/static/www/views/main_scan.html [hunk 4]
```diff
@@ -22,7 +22,7 @@ <h1 class="title">{{'Inventory' | translate}} {{data.inventory.name}}</h1>
 
                 <label class="item item-input" toggle-anim="data.location">
                     <div class="form-field">{{'Current Location' | translate}}</div>
-                    <span class="form-label">{{data.location.name}}</span>
+                    <span class="form-label">{{ data.location.parent_complete_name }} / {{data.location.name}}</span>
```
Comment (legalsylvain) [inline mobile_app_inventory/static/www/views/main_scan.html]: abstain on that point. For me, it depends of how location are named.

Pro to display parent complete name : sometime the name of the location is not sufficient to know exactly on wich location we are working. 
Pro to hide parent complete name : the display can be ugly (very big name), on small devices.
if parent complete name is very big, is the location name hidden ? if no, :+1:

**Review by legalsylvain — COMMENTED** [review]
comment inside. regards.

File: mobile_app_inventory/static/www/views/main_scan.html (hunk 4 above)
Comment (hparfr) [reply mobile_app_inventory/static/www/views/main_scan.html]: What do you suggest then ?

Comment (legalsylvain) [reply mobile_app_inventory/static/www/views/main_scan.html]: maybe 
```
{{data.location.name}} ({{data.location.parent_complete_name}})
```

well location.name should be display in any case. Don't you think ?

**Comment by legalsylvain** [comment]
Hi @hparfr. thanks for the update. LGTM to me except the default url that was generating errors when I was using inventory instead of login.

**Comment by hparfr** [comment]
>LGTM to me except the default url that was generating errors when I was using inventory instead of login.

Let's do it that way. Didn't succeed to reproduce the issue you mention about routing. We can always fix it later if the problem is still laying around. Setting login as the default page is not a reliable fix. It confuses users by prompting credentials to users whom are already logged-in.

**Pull request merged**